packages/compass-crud/src/stores/crud-store.js

```diff
@@ -926,40 +926,18 @@ const configureStore = (options = {}) => {
           });
         }
 
-        // check if the newly inserted document matches the current filter, by
-        // running the same filter but targeted only to the doc's _id.
-        const filter = Object.assign({}, this.state.query.filter, { _id: doc._id });
-        this.dataService.count(this.state.ns, filter, {}, (err, count) => {
-          if (err) {
-            return this.setState({
-              insert: this.getInitialInsertState()
-            });
-          }
-          // track mode for analytics events
-          const payload = {
-            ns: this.state.ns,
-            view: this.state.view,
-            mode: this.state.insert.jsonView ? 'json' : 'default',
-            multiple: false,
-            docs: [doc],
-          };
-          this.localAppRegistry.emit('document-inserted', payload);
-          this.globalAppRegistry.emit('document-inserted', payload);
-
-          // count is greater than 0, if 1 then the new doc matches the filter
-          if (count > 0) {
-            return this.setState({
-              docs: this.state.docs.concat([ new HadronDocument(doc) ]),
-              count: this.state.count + 1,
-              end: this.state.end + 1,
-              insert: this.getInitialInsertState()
-            });
-          }
-          this.setState({
-            count: this.state.count + 1,
-            insert: this.getInitialInsertState()
-          });
-        });
```

Comment:
Are we good with removing this “smart” behavior even for non-CSFLE/QE inserts? I don’t mind, I think it adds consistency and has no major downsides, but I think it’s worth calling out if we do it this way.

Reply:
But how this is different from counting asynchronously while refreshing a list of documents? I mean visually it still shows the proper number, or I miss something?

Reply:
Yeah, visually the end result should be the same, it’s just that refreshing is a more expensive operation than a count (presumably, otherwise this would already have always been refreshing to begin with).

Reply:
True. What do you have in mind by `calling out`? Like our conversation here in PR? :) or some other place like comment/ticket?

Reply:
Yeah, I’m not sure, maybe just merge this and mention it during standup or sth? :)